src/v0/api_token.rs

```diff
@@ -0,0 +1,81 @@
+use crate::db::{
+    models::api_token,
+    schema::{api_tokens, api_tokens::dsl::api_tokens as api_tokens_dsl},
+    DBConnectionPool,
+};
+use crate::v0::{context::SharedContext, errors::HandleError};
+use diesel::query_dsl::{filter_dsl::FilterDsl, RunQueryDsl};
+use diesel::{ExpressionMethods, OptionalExtension};
+use warp::{Filter, Rejection};
+
+/// Header where token should be present in requests
+const API_TOKEN_HEADER: &str = "API-Token";
+
+/// API Token wrapper type
+#[derive(PartialEq, Eq)]
+pub struct APIToken(String);
```

Comment:
For the API Token, us a `[u8; 32]`. We will figure out the crypto for this later. In the header, indeed, encode it in base64 URL